neuro_data/pipeline_management/NeuroDataPipelineManagement.py

```diff
@@ -0,0 +1,181 @@
+import numpy as np
+import datajoint as dj
+
+
+from neuro_data.static_images.data_schemas import StaticScanCandidate, StaticScan, ConditionTier, Frame, InputResponse, Eye, Treadmill, StaticMultiDataset, StaticMultiDatasetGroupAssignment, ExcludedTrial
+
+pipeline_anatomy = dj.create_virtual_module('pipeline_anatomy', 'pipeline_anatomy')
+pipeline_fuse = dj.create_virtual_module('pipeline_fuse', 'pipeline_fuse')
+pipeline_stimulus = dj.create_virtual_module('pipeline_stimulus', 'pipeline_stimulus')
+
+
+class NeuroDataPipelineManagement():
+    def __init__(self):
+        pass
+
+    def manually_insert_area_for_scan(self, target_scan, area):
+        """
+        Give a target_scan, it will label all neurons in there with the given area
+
+        Parameters:
+        target_scan : dict(animal_id, session, scan_idx)
+        area : str
+
+        Returns:
+        None
+        """
+        neuron_unit_keys = (pipeline_fuse.ScanSet().Unit() & target_scan).fetch('KEY')
+        for neuron_unit_key in neuron_unit_keys:
+            neuron_unit_key['brain_area'] = area
+            pipeline_anatomy.AreaMembership().insert1(neuron_unit_key, allow_direct_insert=True)
+
+    def manually_insert_layer_for_scan(self, target_scan, layer):
+        """
+        Give a target_scan, it will label all neurons in there with the given layer
+
+        Parameters:
+        target_scan : dict(animal_id, session, scan_idx)
+        layer : str
+
+        Returns:
+        None
+        """
+        neuron_unit_keys = (pipeline_fuse.ScanSet().Unit() & target_scan).fetch('KEY')
+        for neuron_unit_key in neuron_unit_keys:
+            neuron_unit_key['layer'] = layer
+            pipeline_anatomy.LayerMembership().insert1(neuron_unit_key, allow_direct_insert=True)
+
+    # populate functions
+    def process_static_scan(self, target_scan):
+        """
+        Function that goes and check for every table that needs to be populate as well as provide an option
+        to manaully populate AreaMembership and LayerMembership, assuming that all the neurons can be label the same Area and Layer
+        if not, they the user should manually do it.
+
+        Please refer to neuro_data/notebooks/pipeline_management notebook for an example
+
+        Parameters:
+        target_scan : dict(animal_id, session, scan_idx)
+
+        Returns:
+        None
+        """
+        print('Running preprocessing checks for ', target_scan)
+
+        # Check if the scan has been processed completely
+        if (pipeline_fuse.ScanDone() & target_scan).fetch().size == 0:
+            print('[Preprocessing Check]: ' + str(target_scan) + ' Scan has not been processed yet, please look into pipeline for details')
+            return
+        else:
+            print('[Preprocessing Check]: ScanDone Check Passed')
+
+        # Check if neurons area are labeled
+        if len(pipeline_anatomy.AreaMembership() & target_scan) == 0:
```

Comment:
also this can be if pipeline_anatomy.AreaMembership & target_scan:` datajoint relationships implement the boolean interface, it will be false if the relation is empty, true if not

Reply:
Fair point, wasn't sure if python was that smart.